From 7f2fc85991363c22b08470d212ab21d5e2d09fe6 Mon Sep 17 00:00:00 2001
From: Brahma Reddy Battula <brahma@apache.org>
Date: Tue, 23 May 2017 02:39:19 +0800
Subject: [PATCH 2618/2848] HDFS-11849. JournalNode startup failure exception
 should be logged in log file. Contributed By
 Surendra Singh Lilhore.

(cherry picked from commit 9cab42cc797986081fef184748044f1790a4f039)
(cherry picked from commit cecedcf55839f0847f0488bfb708bda2f2ad7346)

Change-Id: I02e76680a96cf82de74d85f04822a7bd5cc114ea
---
 .../hadoop/hdfs/qjournal/server/JournalNode.java   |    9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/qjournal/server/JournalNode.java b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/qjournal/server/JournalNode.java
index ee7618f..ae8bd6a 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/qjournal/server/JournalNode.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/qjournal/server/JournalNode.java
@@ -17,6 +17,8 @@
  */
 package org.apache.hadoop.hdfs.qjournal.server;
 
+import static org.apache.hadoop.util.ExitUtil.terminate;
+
 import java.io.File;
 import java.io.FileFilter;
 import java.io.IOException;
@@ -304,7 +306,12 @@ public void reportErrorOnFile(File f) {
 
   public static void main(String[] args) throws Exception {
     StringUtils.startupShutdownMessage(JournalNode.class, args, LOG);
-    System.exit(ToolRunner.run(new JournalNode(), args));
+    try {
+      System.exit(ToolRunner.run(new JournalNode(), args));
+    } catch (Throwable e) {
+      LOG.error("Failed to start journalnode.", e);
+      terminate(-1, e);
+    }
   }
 
   public void discardSegments(String journalId, long startTxId)
-- 
1.7.9.5

